File: JuneLong/xystring.py
#Result: 100/100
#Python solution
for tc in range(int(input())):
    students = input()
    students = list(students)
    n = len(students)
    count, i = 0, 0
    while i < n-1:
        if students[i] != students[i+1]: 
            count+=1
            i+=2
        else: i+=1
    print(count)

File: JuneLong/xystring.cpp
//Result: 100/100 Accepted
//C++ solution
#include <bits/stdc++.h>
using namespace std;

#define str string
#define lli long long int
int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	
	int t;
	cin >> t;
	
	while(t--)
	{
	    str s;
	    cin >> s;
	    lli count = 0;
	    int index = 0;
	    
	    while(index < s.length()-1)
	    {
	        if(s[index] != s[index+1])
	        {
	            count+=1;
	            index+=2;
	        }
	        else
	        {
	            index+=1;
	        }
	    }
	    cout << count << '\n';
	    
	}
	return 0;
}
